Building Docker HEAD on ARMv7 through "Make" command fails

Building Docker (HEAD) through "Make" command fails, even with Docker 1.9.1 already installed and fully functional.

System info:

```
Linux docker10 4.1.15-5-ARCH #1 SMP Wed Jan 20 14:05:45 MST 2016 armv7l GNU/Linux
```

```
$ docker version && docker info
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.3
 Git commit:   a34a1d5-dirty
 Built:        Fri Jan 22 09:48:28 UTC 2016
 OS/Arch:      linux/arm
```

```
Server:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.3
 Git commit:   a34a1d5-dirty
 Built:        Fri Jan 22 09:48:28 UTC 2016
 OS/Arch:      linux/arm
Containers: 15
Images: 146
Server Version: 1.9.1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 177
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.1.15-5-ARCH
Operating System: Arch Linux ARM (containerized)
CPUs: 4
Total Memory: 923.1 MiB
Name: docker10
Registry: https://index.docker.io/v1/
WARNING: No memory limit support
WARNING: No swap limit support
```


List the steps to reproduce the issue:
1. git clone ....
2. cd docker
3. make

Describe the results you expected:
Makefile invokes currently installed Docker version to build with Dockerfile.armhf.

(As per https://github.com/docker/docker/blob/master/project/ARM.md )

Describe the results you received:

```
$ make
# A few libnetwork integration tests require that the kernel be
# configured with "dummy" network interface and has the module
# loaded. However, the dummy module is not available by default
# on arm images. This ensures that it's built and loaded.
echo "Syncing kernel modules"
Syncing kernel modules
oc-sync-kernel-modules
make: oc-sync-kernel-modules: Command not found
Makefile:84: recipe for target 'build' failed
make: *** [build] Error 127
```

```
make binary
# A few libnetwork integration tests require that the kernel be
# configured with "dummy" network interface and has the module
# loaded. However, the dummy module is not available by default
# on arm images. This ensures that it's built and loaded.
echo "Syncing kernel modules"
Syncing kernel modules
oc-sync-kernel-modules
make: oc-sync-kernel-modules: Command not found
Makefile:84: recipe for target 'build' failed
make: *** [build] Error 127
```

Extra info:
```
$ sudo echo $(sh -c 'source hack/make/.detect-daemon-osarch && echo ${DOCKER_ENGINE_OSARCH:+$DOCKER_CLIENT_OSARCH} $PACKAGE_ARCH')

linux/arm armhf
```

Thanks

--- a/Makefile
+++ b/Makefile
@@ -84,16 +84,6 @@ binary: build
 	$(DOCKER_RUN_DOCKER) hack/make.sh binary
 
 build: bundles
-ifeq ($(DOCKER_OSARCH), linux/arm)
-	# A few libnetwork integration tests require that the kernel be
-	# configured with "dummy" network interface and has the module
-	# loaded. However, the dummy module is not available by default
-	# on arm images. This ensures that it's built and loaded.
-	echo "Syncing kernel modules"
-	oc-sync-kernel-modules
-	depmod
-	modprobe dummy
-endif
 	docker build ${DOCKER_BUILD_ARGS} -t "$(DOCKER_IMAGE)" -f "$(DOCKERFILE)" .
 
 bundles:
@@ -101,8 +91,8 @@ bundles:
 
 cross: build
 	$(DOCKER_RUN_DOCKER) hack/make.sh dynbinary binary cross
-	
-	
+
+
 win: build
 	$(DOCKER_RUN_DOCKER) hack/make.sh win
 

--- a/project/ARM.md
+++ b/project/ARM.md
@@ -28,3 +28,18 @@ So for example in order to build a Docker binary one has to
 1. clone the Docker/Docker repository on an ARM device `git clone git@github.com:docker/docker.git`  
 2. change into the checked out repository with `cd docker`  
 3. execute `make binary` to create a Docker Engine binary for ARM  
+
+## Kernel modules
+A few libnetwork integration tests require that the kernel be
+configured with "dummy" network interface and has the module
+loaded. However, the dummy module may be not loaded automatically.
+
+To load the kernel module permanently, run these commands as `root`.
+
+    modprobe dummy
+    echo "dummy" >> /etc/modules
+
+On some systems you also have to sync your kernel modules.
+
+    oc-sync-kernel-modules
+    depmod

